protocols/imap/src/test/java/org/apache/james/imap/processor/SetAnnotationProcessorTest.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxManager.MailboxCapabilities;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.slf4j.Logger;
+
+import com.google.common.collect.Lists;
+
+public class SetAnnotationProcessorTest {
+
+    private static final String TAG = "TAG";
+    
+    private SetAnnotationProcessor processor;
+
+    private ImapProcessor mockNextProcessor;
+    private MailboxManager mockMailboxManager;
+    private StatusResponseFactory mockStatusResponseFactory;
+    private ImapProcessor.Responder mockResponder;
+    private ImapSession mockImapSession;
+    private MailboxSession mockMailboxSession;
+    
+    private final List<MailboxAnnotation> MAILBOX_ANNOTATIONS = Lists.newArrayList(MailboxAnnotation.newInstance("/private/key", "anyValue"));
+    private final ImapCommand imapCommand = ImapCommand.anyStateCommand("Name");
+    private MailboxSession.User user = mock(MailboxSession.User.class);
+    private StatusResponse okResponse = mock(StatusResponse.class);
+
+    private SetAnnotationRequest request = new SetAnnotationRequest(TAG, imapCommand, ImapConstants.INBOX_NAME, MAILBOX_ANNOTATIONS);
+
+    @Before
+    public void setUp() {
+        setupMockData();
+        processor = new SetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory);
+        verify(mockMailboxManager).hasCapability(eq(MailboxCapabilities.Annotation));
+    }
+
+    @Test
+    public void getImplementedCapabilitiesShouldContainSupportAnnotationWhenMailboxManagerHasAnnotationCapability() {
+        assertThat(processor.getImplementedCapabilities(null)).containsExactly(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    @Test
+    public void getImplementedCapabilitiesShouldNotContainSupportAnnotationWhenMailboxManagerHasNotAnnotationCapability() {
+        when(mockMailboxManager.hasCapability(eq(MailboxCapabilities.Annotation))).thenReturn(false);
+        assertThat(new SetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory).getImplementedCapabilities(null)).isEmpty();
+    }
+    
+    @Test
+    public void processShouldProcessRequestAndCheckMailboxExist() throws Exception {
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(false);
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager).startProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).endProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).mailboxExists(inbox, mockMailboxSession);
```

Comment:
we don't care whether it calls mailboxExists or not, we just want to know what the behavior is when it does not exist or when it does exist.